### PR TITLE
[framework] make Image.scale select asset variant

### PR DESCRIPTION
**Pull request by jonahwilliams**
Perhaps I'm misunderstanding the intent of the API, but I must admit I have no idea what this argument should have done before.

Based on my read of the intention of the `ExactAssetImage` image provider, it was intended to allow users to declare an image should be drawn with a particular scale. I don't know what else this could mean besides selecting an asset variant with a dpr that is close to the scale.

Without this functionality, users could instead specify the exact path to a particular asset variant perhaps? That plus the scale would work, but feels odd.

With this change Image.asset('some_path.png', scale: 4.0) would select `4.0x/some_path.png` if it exists, or the closest DPR image. Before this change it would always select the base asset variant.

Fixes https://github.com/gskinnerTeam/flutter-scifi/issues/9

### CODE REVIEW

**Comment by goderbauer** [comment]
> Fixes https://github.com/gskinnerTeam/flutter-scifi/issues/9

FYI, this link 404s. Maybe it's a private repository?

**Comment by jonahwilliams** [comment]
Guess its private!

File: packages/flutter/lib/src/painting/image_resolution.dart [hunk 1]
```diff
@@ -398,3 +407,120 @@ class AssetImage extends AssetBundleImageProvider {
...
+
+
+/// Fetches an image from an [AssetBundle], associating it with the given scale.
+///
+/// This implementation requires an explicit final [assetName] and [scale] on
+/// construction, and ignores the device pixel ratio and size in the
+/// configuration passed into [resolve]. For a resolution-aware variant that
+/// uses the configuration to pick an appropriate image based on the device
+/// pixel ratio and size, see [AssetImage].
+///
```
Comment (goderbauer) [inline packages/flutter/lib/src/painting/image_resolution.dart:419]: Can we document here what happens if no asset variant with that exact scale specified exists?

**Comment by goderbauer** [comment]
After looking at this I share your assessment that `ExactAssetImage` in its current form doesn't make too much sense and this change likely represents what it should do. @Hixie may have some additional historical context. He introduced this in https://github.com/flutter/flutter/commit/2dfdc840b1b10342cecd2c89292e2f4bf6003948, describing it as "a variant of AssetImage that doesn't do anything clever".

**Comment by goderbauer** [comment]
(triage) @jonahwilliams Do you still have plans for this PR?

**Pull request closed without merging**